WebDAV sync: 404 until I create file manually

<!--- Your issue may already be reported!
!!! Please search the issues before creating one !!! -->

### Your Environment

<!--- Include as many relevant details about the environment you experienced the bug in -->

- Version used: deb 7.9.2
- Operating System and version: elementary OS 5.1.7 Hera

### Expected Behavior

When creating a webdav sync, the file should be created automatically.

### Current Behavior

I entered the sync file path, but I get `Unknown Sync Error: Invalid response: 404`

### Steps to Reproduce (for bugs)
1. Add webdav sync and enter a sync file path, e.g. `/sync/super-productivity.json`
2. Click save
3. Click button to sync manually
4. **See 404 error as snackbar**
5. create empty file on nextcloud: `/sync/super-productivity.json`
6. Sync manually
7. See dialog that asks if you want to upload data to remote
8. Confirm
9. Works

### Console Output

```
zone.js:2863 HEAD https://my.cloud/remote.php/dav/files/my-user/Backup/me/super-productivity.json 404
d @ zone.js:2863
scheduleTask @ zone.js:393
[...]
web-dav-sync.service.ts:55 Error: Invalid response: 404 
    at Gt (index.js:16)
    at Ye (index.js:16)
    at T.invoke (zone.js:372)
    at Object.onInvoke (core.mjs:25450)
    at T.invoke (zone.js:371)
    at I.run (zone.js:134)
    at zone.js:1276
    at T.invokeTask (zone.js:406)
    at Object.onInvokeTask (core.mjs:25437)
    at T.invokeTask (zone.js:405)
```

### Error Log (Desktop only)

logfile is not printing anything (how can I configure loglevel?)

--- a/src/app/imex/sync/web-dav/web-dav-api.service.ts
+++ b/src/app/imex/sync/web-dav/web-dav-api.service.ts
@@ -3,18 +3,21 @@ import { Observable } from 'rxjs';
 import { GlobalConfigService } from '../../../features/config/global-config.service';
 import { DataInitService } from '../../../core/data-init/data-init.service';
 import { first, map, switchMap, tap } from 'rxjs/operators';
-// @ts-ignore
-import { createClient } from 'webdav/web';
+
 import { WebDavHeadResponse } from './web-dav.model';
 import { IS_ANDROID_WEB_VIEW } from '../../../util/is-android-web-view';
 import { androidInterface } from '../../../features/android/android-interface';
 import { IncomingHttpHeaders } from 'http';
 
-// const createClient = (...args: any) => ({
-//   stat: async (a: any): Promise<any> => undefined,
-//   getFileContents: async (a: any, b?: any): Promise<any> => undefined,
-//   putFileContents: async (a: any, b?: any): Promise<any> => undefined,
-// });
+// Get the Type, but keep the actual bundle payload out
+import type { createClient } from 'webdav';
+type CreateClientType = typeof createClient;
+
+// Lazyload the webdav bundle (when requested)
+const LazyWebDavCreateClient = (): Promise<CreateClientType> =>
+  import(/* webpackChunkName: "webdav-web" */ 'webdav/web').then(
+    ({ createClient }) => createClient as CreateClientType,
+  );
 
 interface AndroidHttpResponse {
   data: string;
@@ -60,10 +63,19 @@ export class WebDavApiService {
       first(),
     );
 
+  private _lazyWebDavClientCache: Promise<CreateClientType> | null = null;
+
   constructor(
     private _globalConfigService: GlobalConfigService,
     private _dataInitService: DataInitService,
   ) {}
+
+  private getWebDavClientCreator(): Promise<CreateClientType> {
+    return (
+      this._lazyWebDavClientCache ??
+      (this._lazyWebDavClientCache = LazyWebDavCreateClient())
+    );
+  }
 
   private checkErrorAndroid(result: AndroidHttpResponse): void {
     if (result.status < 200 || result.status > 299) {
@@ -90,12 +102,14 @@ export class WebDavApiService {
       )) as AndroidHttpResponse;
       this.checkErrorAndroid(result);
     } else {
-      const client = createClient(cfg.baseUrl, {
+      const webDavClientCreator = await this.getWebDavClientCreator();
+
+      const client = webDavClientCreator(cfg.baseUrl, {
         username: cfg.userName,
         password: cfg.password,
       });
 
-      return await client.putFileContents(path, JSON.stringify(data), {
+      await client.putFileContents(path, JSON.stringify(data), {
         contentLength: false,
       });
     }
@@ -116,7 +130,9 @@ export class WebDavApiService {
       this.checkErrorAndroid(result);
       return result.headers as WebDavHeadResponse;
     } else {
-      const client = createClient(cfg.baseUrl, {
+      const webDavClientCreator = await this.getWebDavClientCreator();
+
+      const client = webDavClientCreator(cfg.baseUrl, {
         username: cfg.userName,
         password: cfg.password,
       });
@@ -146,7 +162,9 @@ export class WebDavApiService {
       this.checkErrorAndroid(result);
       return JSON.parse(result.data);
     } else {
-      const client = createClient(cfg.baseUrl, {
+      const webDavClientCreator = await this.getWebDavClientCreator();
+
+      const client = webDavClientCreator(cfg.baseUrl, {
         username: cfg.userName,
         password: cfg.password,
       });

--- a/src/app/imex/sync/web-dav/web-dav-sync.service.ts
+++ b/src/app/imex/sync/web-dav/web-dav-sync.service.ts
@@ -46,11 +46,13 @@ export class WebDavSyncService implements SyncProviderServiceInterface {
         clientUpdate: d.getTime(),
         rev: this._getRevFromMeta(meta),
       };
-    } catch (e: unknown) {
-      const isAxiosError = !!(e && (e as any).response && (e as any).response.status);
-      if (isAxiosError && (e as any).response.status === 404) {
+    } catch (e: any) {
+      const isAxiosError = !!(e?.response && e.response.status);
+
+      if ((isAxiosError && e.response.status === 404) || e.status === 404) {
         return 'NO_REMOTE_DATA';
       }
+
       console.error(e);
       return e as Error;
     }

